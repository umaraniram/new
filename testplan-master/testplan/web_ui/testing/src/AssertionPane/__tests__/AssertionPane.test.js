import React from 'react';
import {shallow} from 'enzyme';
import {StyleSheetTestUtils} from "aphrodite";

import AssertionPane from '../AssertionPane';

function defaultProps() {
  return {
    assertions: [
      {
        "category":"DEFAULT",
        "machine_time":"2019-02-12T17:41:42.795536+00:00",
        "description":null,
        "line_no":25,
        "label":"==",
        "second":"foo",
        "meta_type":"assertion",
        "passed":true,
        "type":"Equal",
        "utc_time":"2019-02-12T17:41:42.795530+00:00",
        "first":"foo"
      }
    ],
    left: "19.5em",
    testcaseUid: "22758cc5-8a89-472b-bf67-b64dbc2c0b40"
  };
}


describe('AssertionPane', () => {
  let props;
  let shallowComponent;

  beforeEach(() => {
    // Stop Aphrodite from injecting styles, this crashes the tests.
    StyleSheetTestUtils.suppressStyleInjection();
    props = defaultProps();
    shallowComponent = undefined;
  });

  it('shallow renders the correct HTML structure', () => {
    shallowComponent = shallow(<AssertionPane {...props}/>);
    expect(shallowComponent).toMatchSnapshot();
  });
});