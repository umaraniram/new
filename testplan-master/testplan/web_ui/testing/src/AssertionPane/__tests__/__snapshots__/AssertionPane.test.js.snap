// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AssertionPane shallow renders the correct HTML structure 1`] = `
<div
  style={
    Object {
      "height": "calc(100% - 5em)",
      "left": "19.5em",
      "paddingLeft": "20px",
      "position": "absolute",
      "top": "5em",
      "width": "calc(100% - 19.5em)",
    }
  }
>
  <div
    className="buttonsDiv_140n9qh"
  >
    <FontAwesomeIcon
      border={false}
      className="icon_1rr1rr0"
      fixedWidth={false}
      flip={null}
      icon="plus-circle"
      inverse={false}
      key="faPlusCircle"
      listItem={false}
      mask={null}
      onClick={[Function]}
      pull={null}
      pulse={false}
      rotation={null}
      size="1x"
      spin={false}
      symbol={false}
      title="Expand all"
      transform={null}
    />
    <FontAwesomeIcon
      border={false}
      className="icon_1rr1rr0"
      fixedWidth={false}
      flip={null}
      icon="minus-circle"
      inverse={false}
      key="faMinusCircle"
      listItem={false}
      mask={null}
      onClick={[Function]}
      pull={null}
      pulse={false}
      rotation={null}
      size="1x"
      spin={false}
      symbol={false}
      title="Collapse all"
      transform={null}
    />
  </div>
  <div
    className="infiniteScrollDiv_4z6319"
  >
    <InfiniteScroll
      items={
        Array [
          Object {
            "category": "DEFAULT",
            "description": null,
            "first": "foo",
            "label": "==",
            "line_no": 25,
            "machine_time": "2019-02-12T17:41:42.795536+00:00",
            "meta_type": "assertion",
            "passed": true,
            "second": "foo",
            "type": "Equal",
            "utc_time": "2019-02-12T17:41:42.795530+00:00",
          },
        ]
      }
      key="22758cc5-8a89-472b-bf67-b64dbc2c0b40"
    >
      <DescriptionPane />
      <AssertionGroup
        entries={Array []}
        resetGlobalIsOpen={[Function]}
      />
      <LogGroup />
    </InfiniteScroll>
  </div>
</div>
`;
